Update the guide for deploying Metabase in Elastic Beanstalk

dragonsahead opened the pull request:
Update the documentation for the ElasticBeanstalk deployment:
- updated all the images for the main deployment flow with current ones
- built new LaunchNow URL's with updated instance types (t3a) and and gave options to all regions
- updated the steps with mandatory VPC creation (otherwise the deployment fails)
- separate the creation of Elastic Beanstalk stack and RDS creation in separate flows
- update the Load Balancers health checks

This should encourage RDS / Elastic Beanstalk decoupling from the beginning and provide a better experience when doing the step by step guide as it works straight away: you don't face errors for not creating VPCs or don't waste time searching for options that are no longer there.

This PR intends to shed some light and (maybe) close:
https://github.com/metabase/metabase/issues/9021: using an RDS database configured by Elastic Beanstalk is an super bad choice for 2 reasons: the database is created with magnetic disks, and it's tied to the cloudformation template. This new guide recommends going in a different path and creating the RDS database separately.
https://github.com/metabase/metabase/issues/13251: this was a temporary issue for sure, now we don't recommend this path anymore. Same as above
https://github.com/metabase/metabase/issues/1336: will not recommend to use the url but rather download the zip from our link and upload it to Elastic Beanstalk, for some reason AWS does not recognize a Cloudfront S3 origin :shrug: 
https://github.com/metabase/metabase/issues/6962: this guide recommends to use ALB for SSL termination in Elastic Beanstalk, having Jetty doing this is a bad choice as the server will be doing something that concerns to a reverse proxy, we "could" also take out the NGINX proxy that we are using in the .ebextensions deployment
https://github.com/metabase/metabase/issues/12944: fixes this one

dragonsahead committed:
Update the guide for deploying Metabase in ElasticBeanstalk

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+Once inside RDS, click on the `Create database` button
+![RDS Create Database](images/RDSCreateDatabase.png)
+
+On the Create Database menu, you have to select MySQL or PostgreSQL as engine types, as these two are the ones that Metabase support as the Application Database (where Metabase will save all of its configurations). For this example we will choose PostgreSQL on its latest version available in AWS at this time (12.4-R1)
```
```suggestion
On the Create Database menu, under **Chose a database creation method**, select "Standard create". Under **Engine options**, select "PostgreSQL." This application database will store all of Metabase's configurations, dashboards, questions, etc. 
```

Do we need to mention MySQL?

paoliniluis replied:
I changed this part as well in the latest commit and made bullet points for each section

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS Create Database](images/RDSCreateDatabase.png)
+
+On the Create Database menu, you have to select MySQL or PostgreSQL as engine types, as these two are the ones that Metabase support as the Application Database (where Metabase will save all of its configurations). For this example we will choose PostgreSQL on its latest version available in AWS at this time (12.4-R1)
+![RDS Postgres Selection](images/RDSPostgresSelection.png)
```
Do we need this screenshot?

As a rule, try to describe, succinctly, what people need to do, and only then think: is that sufficient, or would the addition of a screenshot significantly improve what I need to communicate?

paoliniluis replied:
wiped a couple of screenshots, you are absolutely right, as this is my first guide I'm being screenshotholic

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+On the Settings section, you need to type unique identifier for your database and a master password that you will need to preserve for configuring it on the Metabase instances later on this guide.
+![RDS Templates Section](images/RDSPostgresSettings.png)
+
+On the instance size section, you can choose any instance size that will fit your needs. As this is the application database for Metabase, the sizing of the RDS instance depends on the number of Metabase instances that will be connected to this database, number of users that are using Metabase at once and the number of questions, dashboards and configurations that are saved.
+![RDS Instance size](images/RDSInstanceSize.png)
```
Can we make any recommendations here re: sizing?

paoliniluis replied:
yup, made it on the latest commit

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+On production deployments, you should be using a Multi-AZ(Availability Zone) cluster, as this will ensure that the database does not goes down in case there is an issue on a single availability zone 
+![RDS MultiAZ](images/RDSMultiAZ.png)
+
+On the Connectivity section, you have to ensure that you are deploying the database in the same VPC as the one you deployed the Metabase instance, otherwise they won't be able to see each other. You have to select to create a VPC security group on the section, as you will need to grant access from Metabase instance to the database on the port that listens for connections.
```
```suggestion
On the Connectivity section, you have to ensure that you are deploying the database in the same VPC as the one you deployed the Metabase instance, otherwise they won't be able to see each other. 
```
The sentence "You have to select to create a VPC security group on the section, as you will need to grant access from Metabase instance to the database on the port that listens for connections" is unclear.

paoliniluis replied:
done! made that change and added a bulleted list on the latest commit

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -65,19 +77,50 @@ You can now go ahead and click `Review and launch`.
+
+#### Enabling VPC
+
+You must enable your Application to exist in a VPC unless you will receive an error when creating it. To use a VPC, head to the Network section in the configuration and click on the `Edit` button:
```
We might want to include a one-liner about what a VPC is (and a link to learn more), and why we need to enable it (beyond just "you'll receive an error").

paoliniluis replied:
done! thank!

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -65,19 +77,50 @@ You can now go ahead and click `Review and launch`.
+
+![Elastic Beanstalk Network section](images/EBNetworkSection.png)
+
+Once inside the Network configuration you need to select the VPC where the Application will exist. If you haven't created one then the default one will suffice.
```
If you haven't created one what?

paoliniluis replied:
described more this part, thanks!

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+On the Additional configuration section you have to specify an `Initial database name` as this will be the database that Metabase will use for all of its configurations. From here you can also configure the backup window in case you need to restore the backups at some point in time.
+![RDS Initial Database](images/RDSInitialDatabase.png)
+
+Once this is done, click on the `Create database` button on the lower right part of the page and wait for the database to be created (this can take several minutes).
```
Once what is done?

paoliniluis replied:
changed it to "When you have completed all these configurations", thanks!

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS DB Identifier](images/RDSDBIdentifier.png)
 
-#### Using Metabase in a VPC
+On the page that appears after you click on the database identifer you will see on the center of the page the Connectivity & Security section that will provide you with the `Endpoint` that you will need to give to Metabase to use in order to connect to this database we just created
+![RDS Connection Data](images/RDSConnectionData.png)
```
```suggestion
On the page that appears after you click on the database identifer, you will see on the center of the page the **Connectivity & Security** section. This section will provide you with the **Endpoint** that Metabase needs to connect to the application database we just created.

![RDS Connection Data](images/RDSConnectionData.png)
```

paoliniluis replied:
thanks! changed it for the latest commit!

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS Connection Data](images/RDSConnectionData.png)
 
-Newer AWS accounts are encouraging the use of Virtual Private Cloud (VPC) for deployments, and in general we think it's simplest to follow that best practice.
+Also, in the Security group rules section, you will see the Security Group that was created but it will have a rule that allows only one IP address to access the database. You need to change this rule to allow access to the ElasticBeanstalk environment.
```
```suggestion
In the **Security group rules** section, you'll see the Security Group that was created, but the group will have a rule that allows only one IP address to access the database. You need to change this rule to allow access to the ElasticBeanstalk environment.
```

paoliniluis replied:
changed for the latest commit!

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS Security Group Rules](images/RDSsecurityGroupRules.png)
 
-If you prefer not to use a VPC, that's fine. But one thing to note is that some EC2 instance types (`t2.*` specifically) are not available outside of a VPC, so if you choose to not use a VPC then make sure and pick appropriate instance types.
+You have to click on the INBOUND rule and on the page that opens click again on the Inbound rules tab on the lower part of the page to configure the rule for the database
+![RDS Security Group Rules](images/RDSInboundRule.png)
```
```suggestion
To configure the rule for the database, select the INBOUND rule, then click on the Inbound rules tab on the lower part of the page.

![RDS Security Group Rules](images/RDSInboundRule.png)
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS Security Group Rules](images/RDSInboundRule.png)
 
-If you are launching your Metabase inside of a VPC you'll now need to check a few boxes to enable your application to work inside your VPC subnets. Click `Modify` on this Network box to set things up.
+When you click on Inbound Rules, you need to click on `Edit Inbound Rules` button that appears on the right side of the section
+![RDS Edit Inbound Rule](images/RDSEditInboundRule.png)
```
```suggestion
When you click on Inbound Rules, you need to click on `Edit Inbound Rules` button that appears on the right side of the section.

![RDS Edit Inbound Rule](images/RDSEditInboundRule.png)
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS Edit Inbound Rule](images/RDSEditInboundRule.png)
 
-![Elastic Beanstalk VPC Entry](images/EBVPCEntry.png)
+On edit page you need to delete the IP address that appears as default and add the security group that the ElasticBeanstalk has (you can easily idenfy it as the Security group name will have the keyword AWSEBSecurityGroup on its name). Once you add this click the `Save rules button`
+![RDS Edit Inbound Rule](images/RDSEditInboundRuleSG.png)
```
```suggestion

On the edit page, you need to delete the IP address that appears as default, then add the security group that the ElasticBeanstalk has (the Security group name will have the keyword AWSEBSecurityGroup  in its name). Once you add this security group, click the `Save rules button`.

![RDS Edit Inbound Rule](images/RDSEditInboundRuleSG.png)
```

paoliniluis replied:
thanks!

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
+![RDS Edit Inbound Rule](images/RDSEditInboundRuleSG.png)
 
-Unless you have a custom VPC setup that you know how to configure, it's easiest to just check all the boxes under `Load balancer settings` and `Instance settings` and allow your infrastructure to exist on all subnets. Note that the Load Balancer cannot be in more than one subnet per availability zone, and the database requires subnets in at least two availability zones. Once you've finished your  ...
+Now you can go to the Metabase ElasticBeanstalk deployment and add the RDS instance as the Application Database with Environment variables under the Software configuration
+![EB Environment Variables](images/EBEnvVariables.png)
```
This is a big sentence.

paoliniluis replied:
changed it a bit, thanks!

jeff-bruemmer reviewed (changes requested):
Nice work, Luis!

- Audit the article and see how many screenshots you can cut with zero or minimal information loss. AWS could (and has) changed its UI in the past, so each screenshot incurs a maintenance cost. That and some images are superfluous to the text, which puts an unnecessary tax on the reader.  For example (and this is not your addition), does the source code origin section really merit a before and after screenshot?
- We might want to give a narrative overview up front to give people a big picture view of what we'll be doing. Important to mention the application database.
- Prefer bold text to code formatting for key terms (like section or configuration names). Reserve inline code formatting for code.
- Don't rely on screenshots to communicate configuration. If someone needs to toggle a button in a section, don't just show a screenshot (in fact you may not need the screenshot), but do explicitly write: "In the X section, makes sure Y is set to Z".

Overall, great job, Luis. Hope you find this feedback useful, and keep up the good work!

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,28 +1,44 @@
 
 Metabase provides a pre-configured Elastic Beanstalk launch URL to help you get started with new installations. If you are starting fresh we recommend you follow this link in a new tab to begin creating the Elastic Beanstalk deployment with a few choices pre-filled. Then just follow the step-by-step instructions below to complete your installation.
 
 [Launch Metabase on Elastic Beanstalk](https://downloads.metabase.com/{{ site.latest_version }}/launch-aws-eb.html)
+We will need to update to this link though: https://console.aws.amazon.com/elasticbeanstalk/home?region=us-east-2#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true
```
Nope, we need to replace the launch-aws-eb.html redirection once this is approved

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -65,19 +77,50 @@ You can now go ahead and click `Review and launch`.
+
+### Additional Options
+
+#### Configuring RDS for Metabase INSIDE ElasticBeanstalk (not recommended)
```
I left it just for being consistent with older versions of the documentation

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,28 +1,44 @@
 ![Elastic Beanstalk First Screen](images/EBFirstScreen.png)
 
-_NOTE: If this screenshot doesn't match what you see in the Elastic Beanstalk console, it's likely that you are on an old version of the Elastic Beanstalk UI. At the time of writing this documentation, both versions of the UI are being reported in the wild. If you need to, you can [view our older AWS EB documentation](running-metabase-on-elastic-beanstalk-old.md)._
+_NOTE: If this screenshot doesn't match what you see in the Elastic Beanstalk console, it's likely that you are on an old version of the Elastic Beanstalk UI. If you need to, you can [view our older AWS EB documentation](running-metabase-on-elastic-beanstalk-old.md)._
```
I took it out on the latest commit

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,24 +54,20 @@ As for the domain URL, Feel free to get creative — just remember that the URL
 ![Elastic Beanstalk Base Configuration](images/EBUploadYourCode.png)
 
-The modal should be pre-populated with the correct settings if you used the quick launch URL at the top of these instructions. All you need to do is push the `Upload` button in this modal.
+- On the Source Code Origin section click the `Choose file` button with the `Local File` radio button selected and upload the file you dowloaded previously (metabase-aws-eb.zip):
```
In fact the one that we had is a CloudFront URL that is served 99% sure from a S3 bucket, but IDK why AWS does not recognize it as an S3 URL and returns an error, so I played safe here

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -65,19 +77,50 @@ You can now go ahead and click `Review and launch`.
 
 You will need to enable enhanced health checks for your Beanstalk Monitoring.
```
Added the reason in the latest commit

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
 Once you've entered a password and clicked `Save`, the red outline on the Database box should have gone away, indicating that the application is valid and ready to be launched.
 
-### Additional Options
+#### Configuring RDS for Metabase OUTSIDE ElasticBeanstalk (recommended)
```
I removed this entire section and took it to another page specifically for RDS and the decoupling of Elastic Beanstalk RDS

paoliniluis commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -100,31 +143,68 @@ The database settings screen will give you a number of options for your applicat
-### Additional Options
+#### Configuring RDS for Metabase OUTSIDE ElasticBeanstalk (recommended)
+
+In AWS Console, enter RDS in the search box or select the service from the dropdown button on the top left of the page.
```
I removed the word "Console" from the documentation and replaced it by "page". This dichotomy happens as well in my Cloud Computing classes in the University where my students believe that a "Console" is a terminal in the computer while it's only AWS page while logged-in

dragonsahead committed:
parent bb5f28dee441ca09e610a6918890920592d215a2
author Luis Paolini <paoliniluis@gmail.com> 1610502973 -0300
committer Luis Paolini <paoliniluis@gmail.com> 1614898920 -0300

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+
+# Configuring RDS for Metabase (the recommended guide)
+
+If you want to move from using Metabase just for testing to something that is ready for the big time, you need to use a production-grade database like PostgreSQL or MySQL/MariaDB. This guide will provide you with all the steps for this venture. A high level architecture diagram of what you will achieve once you deploy Metabase with a dedicated application database can be [seen here](images/Metaba ...
```
```suggestion
If you want to move from using Metabase just for testing to something that is ready for the big time, you need to use a production-grade database like PostgreSQL or MySQL/MariaDB. Here's a [high level architecture diagram](images/Metabase-AWS-SI.png) of Metabase deployed with a dedicated application database.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+If you want to move from using Metabase just for testing to something that is ready for the big time, you need to use a production-grade database like PostgreSQL or MySQL/MariaDB. This guide will provide you with all the steps for this venture. A high level architecture diagram of what you will achieve once you deploy Metabase with a dedicated application database can be [seen here](images/Metaba ...
+
+## Step 1
+In AWS, enter RDS in the search box or select the service from the dropdown button on the top left of the page. Once inside RDS, click on the `Create database` button
```
```suggestion
In AWS, enter RDS in the search box or select RDS from the dropdown button on the top left of the page. Once inside RDS, click on the **Create database** button
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+## Step 2
+- Create Database: select MySQL or PostgreSQL as engine types, as these two are the ones that Metabase support as the Application Database (where Metabase will save all of its configurations). For this example we will choose PostgreSQL on its latest version available in AWS at this time (12.4-R1)
+
+- Templates: you can leave `Production` selected or choose any other option that better suits your needs.
```
```suggestion
- Templates: you can leave "Production" selected, or choose any other option that better suits your needs.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+
+- Templates: you can leave `Production` selected or choose any other option that better suits your needs.
+
+- Settings: type a unique `DB instance identifier` for your database. Also remember the username and a master password that you enter here as you will need those to configure the environment variables in Metabase.
```
```suggestion
- Settings: type a unique **DB instance identifier** for your database. You'll need the username and master password to configure the environment variables in Metabase.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+- Settings: type a unique `DB instance identifier` for your database. Also remember the username and a master password that you enter here as you will need those to configure the environment variables in Metabase.
+![RDS Templates Section](images/RDSPostgresSettings.png)
+
+- Instance size: you can choose any instance size that will fit your needs. As this is the application database for Metabase, the sizing of the RDS instance depends on the number of Metabase instances that will be connected to this database, number of users that are using Metabase at once and the number of questions, dashboards and configurations that are saved. To start, a `t3.small` is a good c ...
+![RDS Instance size](images/RDSInstanceSize.png)
```
```suggestion
- Instance size: the sizing of the RDS instance depends on the number of Metabase instances that will be connected to this database, the number of simultaneous users who are using Metabase, and the number of questions, dashboards, and configurations that are saved. To start, a `t3.small` is a good choice.
![RDS Instance size](images/RDSInstanceSize.png)
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+![RDS MultiAZ](images/RDSMultiAZ.png)
+
+- Connectivity: 
+  - ensure that you are deploying the database in the same VPC as the one you deployed the Metabase instance/s, otherwise they won't be able to see each other. 
+  - create a __VPC security group__, as you will need to grant access from the Metabase instance/s to the database on the port that listens for connections.
```
```suggestion
  - Ensure that you are deploying the database in the same VPC as the one you deployed the Metabase instance/s, otherwise they won't be able to see each other. 
  - Create a **VPC security group**, as you will need to grant access from the Metabase instance/s to the database on the port that listens for connections.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+![RDS VPC Security Groups](images/RDSVPCSecurityGroup.png)
+
+- Additional configuration
+  - enter `metabase` as the `Initial database name` as this will be the database that Metabase will use for all of its configurations. 
+  - you can also configure the backup window in case you need to restore the backups at some point in time.
+![RDS Initial Database](images/RDSInitialDatabase.png)
```
```suggestion
  - Enter `metabase` as the **Initial database name**. Metabase will use this database for all of its configurations. 
  - You can also configure the backup window in case you need to restore the backups at some point in time.
![RDS Initial Database](images/RDSInitialDatabase.png)
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+  - you can also configure the backup window in case you need to restore the backups at some point in time.
+![RDS Initial Database](images/RDSInitialDatabase.png)
+
+When you have completed all these configurations, click on the `Create database` button on the lower right part of the page and wait for the database to be created (this can take several minutes).
```
```suggestion
When you've completed all these configurations, click on the **Create database** button on the lower right part of the page and wait for the database to be created (which can take several minutes).
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+
+![RDS Security Group Rules](images/RDSInboundRule.png)
+
+When you click on Inbound Rules, you need to click on `Edit Inbound Rules` button that appears on the right side of the section
```
```suggestion
When you click on Inbound Rules, you need to click on **Edit Inbound Rules** button that appears on the right side of the section.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+
+![RDS Edit Inbound Rule](images/RDSEditInboundRule.png)
+
+On the edit page, you need to delete the IP address that appears as default, then add the security group that the Elastic Beanstalk has (the Security group name will have the keyword AWSEBSecurityGroup  in its name). Once you add this security group, click the `Save rules button`.
```
```suggestion
On the edit page, you need to delete the IP address that appears as default, then add the security group that the Elastic Beanstalk has (the Security group name will have the keyword AWSEBSecurityGroup  in its name). Once you add this security group, click the **Save rules** button.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+![RDS Edit Inbound Rule](images/RDSEditInboundRuleSG.png)
+
+# Step 4 
+After having finished all the previous steps, now you can go to the your Elastic Beanstalk deployment and add the RDS instance as the Application Database with [Environment variables](environment-variables.html) under the [Software configuration](running-metabase-on-elastic-beanstalk.html#set-or-change-environment-variables)
```
```suggestion

After having finished all the previous steps,go to the your Elastic Beanstalk deployment and add the RDS instance as the Application Database with [Environment variables](environment-variables.html) under the [Software configuration](running-metabase-on-elastic-beanstalk.html#set-or-change-environment-variables).
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+---
+# Decouple your RDS database from the Elastic Beanstalk deployment
+
+In the previous versions of this guide, we included the creation and link of a database by default. While this was an easier approach to simplify the deployments, there are many users who can't easily manage the RDS database as it is deeply coupled with the CloudFormation script that was created and executed once they created their Metabase instance.
```
This paragraph needs more context around RDS, Beanstalk and Cloudformation. How exactly does decoupling make management easier?

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+
+In the previous versions of this guide, we included the creation and link of a database by default. While this was an easier approach to simplify the deployments, there are many users who can't easily manage the RDS database as it is deeply coupled with the CloudFormation script that was created and executed once they created their Metabase instance.
+
+**IMPORTANTE NOTES**: 
+- this procedure will generate downtime so make sure to communicate your users that Metabase will be down till you recreate the environment with the new database.
+- you will need the master username and password for the database you used when you created the Elastic Beanstalk instance
```
```suggestion

- This procedure will generate downtime, so make sure to communicate your users that Metabase will be down while you recreate the environment with the new database.
- You'll need the master username and password for the database you used when you created the Elastic Beanstalk instance.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+## Step 1
+Identify the RDS endpoint that your Elastic Beanstalk is using by going to the configuration of the Environment and finding the endpoint value on the Database section.
+![RDS endpooint](images/EBDatabaseEndpoint.png)
+- If the Retention option is "Create snapshot" you are good to go and you can proceed to delete the whole Elastic Beanstalk environment as a snapshot (backup) of the database will be taken before being deleted
```
```suggestion
- If the Retention option is "Create snapshot", you're good to go. You can delete the whole Elastic Beanstalk environment, because AWS will take a snapshot (backup) of the database before deleting the environment.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+Identify the RDS endpoint that your Elastic Beanstalk is using by going to the configuration of the Environment and finding the endpoint value on the Database section.
+![RDS endpooint](images/EBDatabaseEndpoint.png)
+- If the Retention option is "Create snapshot" you are good to go and you can proceed to delete the whole Elastic Beanstalk environment as a snapshot (backup) of the database will be taken before being deleted
+- In case the Retention option has other value, make sure to go to RDS, identify the database that is being used by the Elastic Beanstalk application, select it and take a snapshot
```
```suggestion
- In case the Retention option has a different value, visit your RDS instance and take a snapshot of the database used by the Elastic Beanstalk application.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+![RDS snapshot](images/RDSTakeSnapshot.png)
+
+## Step 2
+Go to the Elastic Beanstalk Metabase Application, select the running environment and terminate it (make sure that in the confirmation that pops up says that the database will be terminated __with snapshot__)
```
```suggestion
Go to the Elastic Beanstalk Metabase Application, select the running environment, and terminate it. Confirm that the database will be terminated __with snapshot__)
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+Go to the Elastic Beanstalk Metabase Application, select the running environment and terminate it (make sure that in the confirmation that pops up says that the database will be terminated __with snapshot__)
+![Terminate environment](images/EBTerminateEnvironment.png)
+
+This step can take many minutes and may even fail for reasons only known by AWS so you will have to identify through CloudFormation which resources failed to be deleted and delete them yourself.
```
```suggestion
This step can take many minutes. If the deletion fails, you'll have to identify through CloudFormation which resources failed to be deleted and delete them yourself.
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+This step can take many minutes and may even fail for reasons only known by AWS so you will have to identify through CloudFormation which resources failed to be deleted and delete them yourself.
+
+## Step 3
+Head to RDS but this time head to the `Snapshots` option on the left of the page where you should see a Manual Snapshot being listed
+![RDS Snapshots](images/RDSSnapshotsMenu.png)
```
```suggestion
Return to RDS and select the **Snapshots** option on the left of the page. You should see a Manual Snapshot listed
![RDS Snapshots](images/RDSSnapshotsMenu.png)
```

jeff-bruemmer commented on docs/operations-guide/creating-RDS-database-on-AWS.md:
```diff
@@ -0,0 +1,98 @@
+Head to RDS but this time head to the `Snapshots` option on the left of the page where you should see a Manual Snapshot being listed
+![RDS Snapshots](images/RDSSnapshotsMenu.png)
+
+Select that snapshot and click on `Actions` -> Restore Snapshot
```
```suggestion
Select that snapshot and click on **Actions** -> Restore Snapshot
```

jeff-bruemmer reviewed (changes requested):
Nice work, Luis. Just a bunch of little suggestions and a couple of notes:

- The advanced topics page needs a title for the page. The current heading for the page makes it seem like the page just covers logging.
- There's currently no way to navigate to these guides from the table of contents https://www.metabase.com/docs/latest/operations-guide/start.html, so you'll need to add entries for them.

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,34 +1,73 @@
 # Running Metabase on AWS Elastic Beanstalk
 
-The Metabase team runs a number of production installations on AWS using Elastic Beanstalk and currently recommend it as the preferred choice for production deployments. Below is a detailed guide to installing Metabase on Elastic Beanstalk.
+AWS Elastic Beanstalk has been a platform for easily deploying Metabase for a long time, however, we have changed this guide in order to provide better support and prevent users from using features from Elastic Beanstalk that will limit scaling Metabase easily in the future.
```
```suggestion
```
Don't think we need this intro.

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,34 +1,73 @@
+If you want to see a high-level architectural diagram of what you will achieve once you follow this guide, [click here](images/Metabase-AWS-H2.png)
 
-Metabase provides a pre-configured Elastic Beanstalk launch URL to help you get started with new installations. If you are starting fresh we recommend you follow this link in a new tab to begin creating the Elastic Beanstalk deployment with a few choices pre-filled. Then just follow the step-by-step instructions below to complete your installation.
+This architecture is the simplest one you can deploy but it is only prepared to test Metabase. Don't worry, this guide gives you all the information to configure this same architecture into something prepared for production workloads if you want to.
```
```suggestion
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,34 +1,73 @@
+AWS Elastic Beanstalk has been a platform for easily deploying Metabase for a long time, however, we have changed this guide in order to provide better support and prevent users from using features from Elastic Beanstalk that will limit scaling Metabase easily in the future.
 
-### Quick Launch
+The following guide will focus on deploying Metabase with a single instance and the embedded H2 database with the following components:
```
```suggestion
This quick launch setup is intended for testing purposes only, and is not intended for production use. We'll focus on deploying Metabase with a single instance and the embedded H2 database with the following components:
```
Let's let them know up front it's not for production.

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,34 +1,73 @@
-After clicking the launch URL, you should see a screen that looks like this:
+## Quick Launch
+
+First of all you need to [download the file](https://downloads.metabase.com/{{ site.latest_version }}/launch-aws-eb.html) to upload to Elastic Beanstalk. This file contains the definition of Metabase docker image
```
```suggestion
Download the [Metabase Docker image](https://downloads.metabase.com/{{ site.latest_version }}/launch-aws-eb.html) to upload to Elastic Beanstalk.
```

Is that link correct?

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,34 +1,73 @@
+
+First of all you need to [download the file](https://downloads.metabase.com/{{ site.latest_version }}/launch-aws-eb.html) to upload to Elastic Beanstalk. This file contains the definition of Metabase docker image
+
+Metabase provides several pre-configured Elastic Beanstalk launch URL to help you get started with new installations. If you are starting fresh we recommend you follow this link in a new tab to begin creating the Elastic Beanstalk deployment with a few choices pre-filled. Then just follow the step-by-step instructions below to complete your installation. Choose your region based on the proximity  ...
```
```suggestion
Metabase provides several pre-configured Elastic Beanstalk launch URLs to help you get started. Open one of the links below in a new tab to create an Elastic Beanstalk deployment with a few choices pre-filled. Then just follow the step-by-step instructions below to complete your installation. 

Choose your region based on the proximity of your users, or if you have strict regulatory requirements that don't let you spin up servers in other countries:
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+While most of the fields here will be correctly pre-filled by following the launch URL above, you'll just need to do two things:
 
-![ebnewenv](images/EBWebTier.png)
+- First, make sure `Platform` is set to `Docker`, with the platform branch dropdown set to `Docker running on 64bit Amazon Linux 2` and Platform version to the one that has a `(Recommended)` tag.
+- Next, change the `Application code` setting to `Upload your code`
```
```suggestion
1. Make sure **Platform** is set to `Docker`, with the platform branch dropdown set to `Docker running on 64bit Amazon Linux 2`, and Platform version to the one that has a `(Recommended)` tag.
2. Change the **Application code** setting to `Upload your code`.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
 
-- First, make sure `Platform` is set to `Preconfigured platform`, with the dropdown set to `Docker`.
-- Next, change the `Application code` setting to `Upload your code`, then click the `Upload` button. You'll see a modal like this:
+These settings will run the Metabase application using [Docker](https://www.docker.com) under the hood, using the official Metabase Docker image which is [published on Dockerhub](https://hub.docker.com/r/metabase/metabase/).
```
```suggestion
These settings will run the Metabase application using the [official Metabase Docker image on Dockerhub](https://hub.docker.com/r/metabase/metabase/).
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+These settings will run the Metabase application using [Docker](https://www.docker.com) under the hood, using the official Metabase Docker image which is [published on Dockerhub](https://hub.docker.com/r/metabase/metabase/).
 
-![Elastic Beanstalk Base Configuration](images/EBUploadYourCode.png)
+You can now go ahead and click `Review and launch` and you will be directed to a page with many sections where you will have to do a bit more configuration to launch your instance.
```
```suggestion
Click **Review and launch**.  You'll be directed to a page to configure and launch your instance.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+### 2.1 Enabling enhanced health checks
 
-You can now go ahead and click `Review and launch`.
+You will need to enable enhanced health checks for your Elastic Beanstalk environment as this will tell the load balancer that the environment is healthy.
```
```suggestion
To set up your load balancer, you'll need to enable enhanced health checks for your Elastic Beanstalk environment. 
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+![Elastic Beanstalk Monitoring](images/EBLoadBalancerEdit.png)
 
-Click on the `Modify` link under the Monitoring section as seen here:
+In the Processes section, select the default process and click on `Actions` -> Edit
```
```suggestion
In the Processes section, select the default process and click on `Actions` -> Edit.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+![Elastic Beanstalk Monitoring Process](images/EBProcessesSection.png)
 
-The `Health check path` is how Elastic Beanstalk knows when the application is ready to run. This path is also used by the Load Balancer to check application heath. Set this path to `/api/health`.
+The `Health check path` is where the Load balancer asks the application if its healthy so it can send traffic to. Set this path to `/api/health`
```
```suggestion
The **Health check path** is where the Load balancer asks the application if its healthy so it can send traffic to. Set this path to `/api/health`
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+### 2.2 Enabling VPC
 
-To run Metabase in a cloud environment of any kind we highly recommend using an independent database server with high availability such as Amazon RDS. So for standard deployments we will choose to create an RDS instance with our Elastic Beanstalk application.
+A Virtual Private Cloud (VPC) is a virtual network that will be logically isolated for the resources that you put inside. Inside these VPC's you can create subnets, firewall rules, route tables and many more. It's one of the foundational features of AWS and you can learn more about it [here](https://aws.amazon.com/vpc/faqs/)
```
```suggestion
A Virtual Private Cloud (VPC) is a virtual network you can use to isolate resources. Inside these VPC's, you can create subnets, firewall rules, route tables and many more. It's one of the foundational features of AWS, and you can learn more about it [here](https://aws.amazon.com/vpc/faqs/)
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+A Virtual Private Cloud (VPC) is a virtual network that will be logically isolated for the resources that you put inside. Inside these VPC's you can create subnets, firewall rules, route tables and many more. It's one of the foundational features of AWS and you can learn more about it [here](https://aws.amazon.com/vpc/faqs/)
 
-_NOTE: It's possible to skip this step if you wish. However, this will force Metabase to use a local H2 database file on your application server, and there will be no way to backup and maintain that database. **When your instance is restarted for any reason you'll lose all your Metabase data**. If you are just doing a quick trial of Metabase that may be okay, but otherwise we recommend against it ...
+You must enable your Application to exist in a VPC, otherwise you will receive an error when creating it. To use a VPC, head to the Network section in the configuration and click on the `Edit` button.
```
```suggestion
You must enable your Application to exist in a VPC, otherwise you'll receive an error when creating it. To use a VPC, head to the **Network** section in the configuration and click on the `Edit` button.
```

walterl replied:
```suggestion
You must configure your Application to launch in a VPC, otherwise you will receive an error when creating it. To use a VPC, head to the Network section in the configuration and click on the `Edit` button.
```

> ... otherwise you will receive an error when creating it.

This doesn't seem accurate. What error will I receive, and why? Is a VPC a requirement? If so, rather say that.

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+![Elastic Beanstalk Network section](images/EBNetworkSection.png)
 
-![Elastic Beanstalk Database Configuration Options](images/EBDatabaseConfigurationOptions.png)
+Once inside the Network configuration you need to select the VPC where the Application will exist. If you haven't created a VPC, then AWS creates a `default` VPC per Region that you can use.
```
```suggestion
Once inside the Network configuration, you need to select the VPC where the Application will exist. If you haven't created a VPC, then AWS creates a `default` VPC per region that you can use.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+Once inside the Network configuration you need to select the VPC where the Application will exist. If you haven't created a VPC, then AWS creates a `default` VPC per Region that you can use.
 
-The database settings screen will give you a number of options for your application database. Regarding individual settings, we recommend:
+You need to select __at least__ 2 zones where the Load Balancer will balance the traffic and also __at least__ 1 zone where the instance will exist. There has to be a zone in common for the balancer to send traffic to a living instance.
```
```suggestion
You need to select __at least__ 2 zones where the Load Balancer will balance the traffic, and  __at least__ 1 zone where the instance will exist. For the load balancer to send traffic to a living instance, there has to be a zone in common.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+![Elastic Beanstalk Networking configuration](images/EBNetworkingConfig.png)
 
-![Elastic Beanstalk Database Settings](images/EBDatabaseSettings.png)
+After configuring the zones where the Load Balancer will exist and the ones that the application will live, click `Save` at the bottom of the page. 
```
```suggestion
After configuring the zones for both the load balancer and the application, click **Save** at the bottom of the page. 
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+
+### 2.3 Final step and deploy
+
+Now go to the Capacity section and click `Edit`
+![Elastic Beanstalk Networking configuration](images/EBCapacity.png)
```
```suggestion
Now go to the Capacity section and click **Edit**.

![Elastic Beanstalk Networking configuration](images/EBCapacity.png)
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+## PostgreSQL Metabase INSIDE Elastic Beanstalk configuration (not recommended)
 
-If you are launching your Metabase inside of a VPC you'll now need to check a few boxes to enable your application to work inside your VPC subnets. Click `Modify` on this Network box to set things up.
+This was the recommended step in the previous versions of the configuration of Metabase in Elastic Beanstalk, however, as AWS Elastic Beanstalk creates a CloudFormation template when you hit the `Create App` button, this means that the database will be created with the Elastic Beanstalk stack and removed when you remove the application.
```
```suggestion
When you hit the `Create App` button, AWS Elastic Beanstalk creates a CloudFormation template.  This template means that the database will be created with the Elastic Beanstalk stack, and removed when you remove the application.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
 
-There are many ways to customize your Beanstalk deployment, but commonly modified settings include:
+- `Snapshot` should be left as `None`.
+- `Engine` should be set to `postgres`. Metabase also supports MySQL/Maria DB as backing databases but **only** on a separate running RDS connected via [environment variables](#set-or-change-environment-variables). Trying to follow these steps and selecting MySQL will result in an error.
```
```suggestion
- `Engine` should be set to `postgres`. Metabase also supports MySQL/Maria DB as backing databases, but **only** on a separate running RDS connected via [environment variables](#set-or-change-environment-variables). Trying to follow these steps and selecting MySQL will result in an error.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -173,87 +234,4 @@ Here's each step:
-- Scroll to the bottom of the page and click `Apply` in the lower right, then wait for your application to update.
-
-_NOTE: Sometimes these settings will not apply until you restart your application server, which you can do by either choosing `Restart App Server(s)` from the Actions dropdown or by deploying the same version again._
+Once a new version is deployed you can safely delete the old Application Version if desired. We recommend keeping at least one previous version available for a while in case you desire to revert for any reason.
```
```suggestion
Once a new version is deployed, you can safely delete the old Application Version. We recommend keeping at least one previous version available for a while in case you want to revert for any reason.
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -136,21 +197,21 @@ When you click `Next` a new tab will open in your browser and you will be prompt
+## Set or change environment variables
 
-When it's all done you should see something like this:
+In order to configure environment variables for your Elastic Beanstalk deployment, (e.g: [to connect the deployment to a separate RDS database](creating-RDS-database-on-AWS.html)) click on your Metabase environment in Elastic Beanstak, go to Configuration -> Software and you will find Environment Properties in the bottom. 
```
```suggestion
In order to configure environment variables for your Elastic Beanstalk deployment (e.g., [to connect the deployment to a separate RDS database](creating-RDS-database-on-AWS.html)), click on your Metabase environment in Elastic Beanstak, go to Configuration -> Software, and look for the Environment Properties in the bottom. 
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -136,21 +197,21 @@ When you click `Next` a new tab will open in your browser and you will be prompt
+In order to configure environment variables for your Elastic Beanstalk deployment, (e.g: [to connect the deployment to a separate RDS database](creating-RDS-database-on-AWS.html)) click on your Metabase environment in Elastic Beanstak, go to Configuration -> Software and you will find Environment Properties in the bottom. 
 
-![ebcomplete](images/EBComplete.png)
+From here you will be able to set or change the variables for configuring many properties of how [your Metabase deployment works](https://metabase.com/docs/latest/operations-guide/environment-variables.html)
```
```suggestion

In the Environment Properties section, you'll be able to set or change the variables for [configuring your Metabase deployment](https://metabase.com/docs/latest/operations-guide/environment-variables.html)
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -136,21 +197,21 @@ When you click `Next` a new tab will open in your browser and you will be prompt
+## Notifications
 
----
+Enter an `Email address` (`Notifications` block) to get notifications about your deployments and changes to your application. This is a very simple way to keep tabs on your Metabase environment, so we recommend putting a valid email in here.
```
```suggestion
For a simple way to keep tabs on your application, enter an email address  in the **Notifications** block to get notifications about your deployment and any changes to your application.. 
```

jeff-bruemmer commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+- `Engine version` can simply be left on the default, which should be the latest version.
+- For `Instance class` you can choose any size, but we recommend `db.t2.small` or larger for production installs. Metabase is pretty efficient so there is no need to make this a big instance.
+- You can safely leave `Storage` to the default size.
+- Pick a `Username` and `Password` for your database. We suggest you hold onto these credentials in a password manager, as it can be useful for things like backups or troubleshooting. These settings will be automatically made available to your Metabase instance, so you will not need to put them in anywhere manually.
```
```suggestion
- Pick a `Username` and `Password` for your database. We suggest you hold onto these credentials in a password manager, as they can be useful for things like backups or troubleshooting. These settings will be automatically made available to your Metabase instance.
```

jeff-bruemmer reviewed (changes requested):
Some suggestions, mostly line edits.

Thanks for putting this update together, Luis. Nice work!

paoliniluis commented:
@robdaemon, @walterl or @mestes , if you want to check it out from the technical standpoint it would be awesome :)

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -1,34 +1,73 @@
+
+Choose your region based on the proximity of your users, or if you have strict regulatory requirements that don't let you spin up servers in other countries:
+
+- US-East-1 [North Virginia](https://console.aws.amazon.com/elasticbeanstalk/home?region=us-east-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- US-East-2 [Ohio](https://console.aws.amazon.com/elasticbeanstalk/home?region=us-east-2#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- US-West-1 [North California](https://console.aws.amazon.com/elasticbeanstalk/home?region=us-west-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- US-west-2 [Oregon](https://console.aws.amazon.com/elasticbeanstalk/home?region=us-west-2#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- CA-Central-1 [Canada](https://console.aws.amazon.com/elasticbeanstalk/home?region=ca-central-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- EU-north-1 [Stockholm](https://console.aws.amazon.com/elasticbeanstalk/home?region=eu-north-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- EU-West-3 [Paris](https://console.aws.amazon.com/elasticbeanstalk/home?region=eu-west-3#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- EU-West-2 [London](https://console.aws.amazon.com/elasticbeanstalk/home?region=eu-west-2#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- EU-West-1 [Ireland](https://console.aws.amazon.com/elasticbeanstalk/home?region=eu-west-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- EU-Central-1 [Frankfurt](https://console.aws.amazon.com/elasticbeanstalk/home?region=eu-central-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- AP-South-1 [Mumbai](https://console.aws.amazon.com/elasticbeanstalk/home?region=ap-south-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- AP-Northeast-1 [Tokyo](https://console.aws.amazon.com/elasticbeanstalk/home?region=ap-northeast-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- AP-Northeast-2 [Seoul](https://console.aws.amazon.com/elasticbeanstalk/home?region=ap-northeast-2#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- AP-Southeast-1 [Singapore](https://console.aws.amazon.com/elasticbeanstalk/home?region=ap-southeast-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- AP-Southeast-2 [Sydney](https://console.aws.amazon.com/elasticbeanstalk/home?region=ap-southeast-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
+- SA-east-1 [São Paulo](https://console.aws.amazon.com/elasticbeanstalk/home?region=sa-east-1#/newApplication?applicationName=Metabase&platform=Docker&environmentType=LoadBalancing&tierName=WebServer&instanceType=t3a.small&withVpc=true)
```
If you're going to list regions, I'd suggest you list all of them. A cursory glance shows you're missing (at least) `af-south-1` (:south_africa:) and `me-south-1`. I also have a weak preference for keeping the code names lower cased, as AWS uses it everywhere.

paoliniluis replied:
Yes, but as af-south-1 and me-south-1 are not enabled by default (you have to enable them from the account settings), I didn't want to produce more friction in the experience, that's why I didn't list those

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+![Elastic Beanstalk Base Configuration](images/EBUploadYourCode.png)
 
-![Elastic Beanstalk Base Configuration](images/EBBaseConfig.png)
+- On the Source Code Origin section click the `Choose file` button with the `Local File` radio button selected and upload the file you dowloaded on the very beginning of this guide (metabase-aws-eb.zip):
```
```suggestion
- In the **Source code origin** section click the `Choose file` button with the `Local File` radio button selected and upload the file you dowloaded at the very beginning of this guide (`metabase-aws-eb.zip`):
```

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+![Elastic Beanstalk Monitoring endpoint](images/EBProcessEditEndpointHealth.png)
 
-![Elastic Beanstalk Monitoring Settings](images/EBMonitoringSettings.png)
+After configuring this health check you can click on `Save` at the bottom of the page
```
```suggestion
After configuring this health check you can click on `Save` at the bottom of the page.
```

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+Now that you’ve installed Metabase, it’s time to [set it up and connect it to your database](../setting-up-metabase.md).
 
-Once you've entered a password and clicked `Save`, the red outline on the Database box should have gone away, indicating that the application is valid and ready to be launched.
+# Additional Options
```
I see this is the original term, but how about changing it to something like _Optional extras_?

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
 
-Newer AWS accounts are encouraging the use of Virtual Private Cloud (VPC) for deployments, and in general we think it's simplest to follow that best practice.
+- `Instance type` (`Instances` block) is for picking the size of AWS instance you want to run. We recommend **at least** `t3a.small` for most uses. You can always [scale](https://www.metabase.com/learn/data-diet/analytics/metabase-at-scale.html) vertically by changing this configuration.
+- `EC2 key pair` (`Security` block) is only needed if you want to SSH into your instance directly which is __not recommended__
```
```suggestion
- `EC2 key pair` (`Security` block) is only needed if you want to SSH into your instance directly which is __not recommended__.
```

Did you mean to use italics for emphasis here? Then it should be a single `_` on either side. If you meant for it to be bold, it's probably best to keep it consistent with other styling (`**`).

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+If you want to use a production-grade database based on best practices to persist all Metabase configurations you have to [create one in RDS separately](creating-RDS-database-on-AWS.html) or manage your own on a separate server and then connect the Elastic Beanstalk instance/s with the RDS database through [environment variables](#set-or-change-environment-variables).
 
-Unless you have a custom VPC setup that you know how to configure, it's easiest to just check all the boxes under `Load balancer settings` and `Instance settings` and allow your infrastructure to exist on all subnets. Note that the Load Balancer cannot be in more than one subnet per availability zone, and the database requires subnets in at least two availability zones. Once you've finished your  ...
+If you want to continue on this path and you know what you are doing, then: look for the Database configuration pane as below. and click on the `Edit` button.
```
```suggestion
If you want to continue on this path and you know what you are doing, then: look for the **Database** configuration pane as below. and click on the `Edit` button.
```

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -38,95 +77,117 @@ As for the domain URL, Feel free to get creative — just remember that the URL
+- `EC2 key pair` (`Security` block) is only needed if you want to SSH into your instance directly which is __not recommended__
 
-If you prefer not to use a VPC, that's fine. But one thing to note is that some EC2 instance types (`t2.*` specifically) are not available outside of a VPC, so if you choose to not use a VPC then make sure and pick appropriate instance types.
+## PostgreSQL Metabase INSIDE Elastic Beanstalk configuration (not recommended)
```
This title reads a bit awkwardly. How about this?

```suggestion
## Application database in Elastic Beanstalk environment (not recommended)
```

Then, I think it would also help to explain what is meant by this approach, before explaining why it's not recommended.

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -136,21 +197,22 @@ When you click `Next` a new tab will open in your browser and you will be prompt
-![ebcomplete](images/EBComplete.png)
 
-To see your new Metabase instance, simply click on the link in parentheses next to your environment name in the top-left. In this example it's `metabase-env-tttt.elasticbeanstalk.com`
+In the Environment Properties section, you'll be able to set or change the variables for [configuring your Metabase deployment](https://metabase.com/docs/latest/operations-guide/environment-variables.html)
```
```suggestion
In the Environment Properties section, you'll be able to set or change the variables for [configuring your Metabase deployment](https://metabase.com/docs/latest/operations-guide/environment-variables.html).
```

walterl commented on docs/operations-guide/running-metabase-on-elastic-beanstalk.md:
```diff
@@ -136,21 +197,22 @@ When you click `Next` a new tab will open in your browser and you will be prompt
+## Notifications
 
-# Deploying New Versions of Metabase
+For a simple way to keep tabs on your application, enter an email address  in the **Notifications** block to get notifications about your deployment and any changes to your application.. 
```
```suggestion
For a simple way to keep tabs on your application, enter an email address  in the **Notifications** block to get notifications about your deployment and any changes to your application.
```

walterl commented on docs/operations-guide/advanced-topics-for-running-Metabase-in-AWS-ElasticBeanstalk.md:
```diff
@@ -0,0 +1,90 @@
+# Logging
+## Network Access log
+
+If you need a log of all the IP addresses and URLs that were accesed during a specific period, you can make the Load Balancer to send those logs to S3. This is useful for analyzing the traffic to your Metabase instance.
```
```suggestion
If you need a log of all the IP addresses and URLs that were accesed during a specific period, you can configure the Load Balancer to send those logs to S3. This is useful for analyzing the traffic to your Metabase instance.
```

walterl commented on docs/operations-guide/advanced-topics-for-running-Metabase-in-AWS-ElasticBeanstalk.md:
```diff
@@ -0,0 +1,90 @@
+
+If you need a log of all the IP addresses and URLs that were accesed during a specific period, you can make the Load Balancer to send those logs to S3. This is useful for analyzing the traffic to your Metabase instance.
+
+To enable this logging, you have to go to the settings of the Load Balancer and enable "Store Logs" in the Access Log Files section. You will need to choose an S3 bucket to dump all the logs and a prefix that will identify the logs coming from this load balancer.
```
```suggestion
To enable this logging, you have to go to the settings of the Load Balancer and enable **Store Logs** in the **Access Log Files** section. You will need to choose an S3 bucket to dump the logs in, and a prefix that will identify the logs coming from this load balancer.
```

walterl commented on docs/operations-guide/advanced-topics-for-running-Metabase-in-AWS-ElasticBeanstalk.md:
```diff
@@ -0,0 +1,90 @@
+If you want to retain the Metabase application logs, you can publish them to an S3 bucket:
+
+- On your Metabase Elastic Beanstalk environment, click on the `Configuration` link in the navigation bar on the left side. On the configuration page, click on the `Edit` button next to `Software`.
+- Scroll down and then check the box next to **Enabled** in S3 log storage section
```
```suggestion
- Scroll down and then check the box next to **Enabled** in S3 log storage section.
```

walterl commented on docs/operations-guide/advanced-topics-for-running-Metabase-in-AWS-ElasticBeanstalk.md:
```diff
@@ -0,0 +1,90 @@
+  - The certificate MUST match the domain you plan to use for your Metabase install.
+- In SSL Policy select "ELBSecurityPolicy-TLS-1-2-2017-01".
+- Scroll to the bottom of the page and click **Save**.
+  - Your Environment will begin updating with your new change. You will have to wait for this to complete before making additional updates.
```
```suggestion

Your Environment will begin updating with your new change. You will have to wait for this to complete before making additional updates.
```

walterl reviewed (approved):
The vast majority of my comments are just minor language/copy tweaks that you should regard as suggestions. It's not my area of expertise, so feel free treat them as optional. :slightly_smiling_face:.

From what I can remember setting up EB apps, there aren't any technical details that strike me as incorrect. As long as you've gone through all the steps, it's good with me :+1:.

dragonsahead committed:
Please work CircleCI
